include/secp256k1.h

```diff
@@ -141,10 +141,10 @@ typedef int (*secp256k1_nonce_function)(
 #   define SECP256K1_API            __declspec (dllexport)
 #   define SECP256K1_API_VAR extern __declspec (dllexport)
 #  endif
-# elif defined _MSC_VER
-#  define SECP256K1_API
-#  define SECP256K1_API_VAR  extern __declspec (dllimport)
-# elif defined DLL_EXPORT
+  /* The user must define SECP256K1_STATIC when consuming libsecp256k1 as a static
```

Comment:
This isn't really a *must*, right? Afaics nothing is changing except adding a new opt-in for dllimport for mingw?

Reply:
@theuni This also removes this branch:
```
# elif defined _MSC_VER
#  define SECP256K1_API
#  define SECP256K1_API_VAR  extern __declspec (dllimport)
```
This was libtool's hack for making the build work on MSVC with static and dynamic linkage and without explicit config by the user. But this comes at the expense of confusing linker warnings with static linkage and other potential issues:

> With Microsoft tools you typically get away with always compiling the code such that variables are expected to be imported from a DLL and functions are expected to be found in a static library. The tools will then automatically import the function from a DLL if that is where they are found. If the variables are not imported from a DLL as expected, but are found in a static library that is otherwise pulled in by some function, the linker will issue a warning (LNK4217) that a locally defined symbol is imported, but it still works. In other words, this scheme will not work to only consume variables from a library. There is also a price connected to this liberal use of imports in that an extra indirection is introduced when you are consuming the static version of the library. That extra indirection is unavoidable when the DLL is consumed, but it is not needed when consuming the static library. 

https://www.gnu.org/software/libtool/manual/html_node/Windows-DLLs.html#Windows-DLLs


@hebasto and I now believe that it's cleaner and simpler to require the user to set `SECP256K1_STATIC` explicitly when static linkage is desired. Such a macro seems to be common on Windows.